### ISSUE TITLE
[Bug]: NEXTCLOUD 26 Email Setup no Test successfully

### ISSUE THREAD

**Issue by apg1980** [nextcloud/server#37329]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After updating to NC26 from NC25 the Email Test ist not working anymore as before.
Beim Senden der E-Mail ist ein Problem aufgetreten. Bitte überprüfe deine Einstellungen. (Fehler: E-Mail konnte nicht versandt werden. Prüfe dein E-Mail-Server-Protokoll)
It worked since years with the same setup.
Debug Log for smtp is enabled but no output.

### Steps to reproduce

Goto Grundeinstellungen Send Test Mail

### Expected behavior

success

### Installation method

Other Community project

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "10.10.20.49",
            "cloud.grunet.de",
            "signaling.grunet.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "tempdirectory": "\/NEXTCLOUD-DATA\/data\/tmp",
        "dbtype": "mysql",
        "version": "26.0.0.11",
        "overwrite.cli.url": "https:\/\/cloud.grunet.de",
        "overwriteprotocol": "https",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "dbindex": 0
        },
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpdebug": true,
        "auth.bruteforce.protection.enabled": false,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "allow_local_remote_servers": true,
        "simpleSignUpLink.shown": false,
        "skeletondirectory": "",
        "auth.webauthn.enabled": false,
        "auto_logout": false,
        "session_keepalive": true,
        "session_lifetime": 86400,
        "remember_login_cookie_lifetime": 1296000,
        "login_form_autocomplete": true,
        "default_language": "de",
        "default_locale": "de_DE",
        "quota_include_external_storage": false,
        "logtimezone": "Europe\/Berlin",
        "loglevel": 2,
        "log_rotate_size": 26214400,
        "filesystem_check_changes": 1,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "appstore.experimental.enabled": false,
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\TXT",
            "OC\\Preview\\PDF",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\Image",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\SVG",
            "OC\\Preview\\Movie",
            "OC\\Preview\\MKV",
            "OC\\Preview\\MP4",
            "OC\\Preview\\AVI",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\BMP",
            "OC\\Preview\\GIF",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\HEIF"
        ],
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "preview_max_x": "1080",
        "preview_max_y": "1920",
        "preview_max_scale_factor": 1,
        "preview_max_memory": 512,
        "activity_expire_days": 90,
        "jpeg_quality": "60",
        "defaultapp": "files",
        "updater.release.channel": "stable",
        "maintenance": false,
        "data-fingerprint": "a5a9ea759f55ea52cd84a318a8071c2e",
        "app.mail.verify-tls-peer": false,
        "app.mail.imap.timeout": 30,
        "app.mail.smtp.timeout": 30,
        "theme": "",
        "knowledgebaseenabled": false,
        "default_phone_region": "DE",
        "ncd_admin_settings": {
            "disallow_aria2_settings": "1"
        },
        "app_install_overwrite": [
            "news",
            "richdocumentscode",
            "fulltextsearch",
            "files_fulltextsearch",
            "fulltextsearch_elasticsearch",
            "passwords",
            "files_fulltextsearch_tesseract",
            "memories"
        ],
        "memories.exiftool": "\/var\/www\/nextcloud\/apps\/memories\/lib\/..\/exiftool-bin\/exiftool-amd64-glibc",
        "blacklisted_files": [
            ".htaccess",
            "Thumbs.db",
            "thumbs.db"
        ],
        "memories.gis_type": 1,
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "memories.vod.path": "\/var\/www\/nextcloud\/apps\/memories\/exiftool-bin\/go-vod-amd64",
        "memories.vod.disable": false
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.18.0
  - admin_audit: 1.16.0
  - calendar: 4.3.0
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - collectives: 2.3.0
  - comments: 1.16.0
  - contacts: 5.2.0
  - contactsinteraction: 1.7.0
  - dav: 1.25.0
  - deck: 1.9.0
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_automatedtagging: 1.16.0
  - files_external: 1.18.0
  - files_fulltextsearch: 25.0.0
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - fulltextsearch: 25.0.0
  - fulltextsearch_elasticsearch: 25.0.0
  - groupfolders: 14.0.0
  - integration_openai: 1.0.2
  - integration_youtube: 0.1.3
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - maps: 1.0.0
  - memories: 4.12.3
  - news: 21.1.0
  - nextcloud_announcements: 1.15.0
  - notes: 4.7.0
  - notifications: 2.14.0
  - notify_push: 0.6.0
  - oauth2: 1.14.0
  - password_policy: 1.16.0
  - passwords: 2023.3.21
  - previewgenerator: 5.2.1
  - provisioning_api: 1.16.0
  - recognize: 3.7.0
  - related_resources: 1.1.0-alpha1
  - richdocuments: 8.0.0
  - richdocumentscode: 22.5.802
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - snappymail: 2.27.1
  - spreed: 16.0.0
  - systemtags: 1.16.0
  - text: 3.7.2
  - theming: 2.1.1
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - user_status: 1.6.0
  - viewer: 1.10.0
  - weather_status: 1.6.0
  - workflowengine: 2.8.0
Disabled:
  - bruteforcesettings: 2.6.0
  - dashboard: 7.6.0 (installed 7.5.0)
  - encryption: 2.14.0 (installed 2.7.0)
  - firstrunwizard: 2.15.0 (installed 2.13.0)
  - metadata: 0.17.0 (installed 0.17.0)
  - ncdownloader: 1.0.2 (installed 1.0.2)
  - photos: 2.2.0 (installed 2.0.1)
  - privacy: 1.10.0 (installed 1.2.0)
  - quota_warning: 1.15.0 (installed 1.15.0)
  - recommendations: 1.5.0 (installed 1.4.0)
  - support: 1.9.0 (installed 1.0.0)
  - survey_client: 1.14.0 (installed 1.4.0)
  - suspicious_login: 4.4.0 (installed 3.1.0)
  - twofactor_totp: 8.0.0-alpha.0 (installed 5.0.0)
  - user_ldap: 1.16.0 (installed 1.15.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
`<html><body>
<!--StartFragment-->

  | Undefined array key "mail_smtpauthtype" at /var/www/nextcloud/apps/settings/templates/settings/admin/additional-mail.php#132
-- | --


<!--EndFragment-->
</body>
</html>`
```

### Additional info

`  'mail_from_address' => 'noreply',
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_domain' => 'domain.de',
  'mail_smtphost' => 'IP',
  'mail_smtpport' => '25',
  'mail_smtpdebug' => true,
`

**Comment by miaulalala** [comment]
Are you using authentication, i. e. is this tickbox checked for you?

![image](https://user-images.githubusercontent.com/7427347/226887940-e7655bd8-1337-4bd8-b30c-64abdd89a368.png)

**Comment by apg1980** [comment]
no i do not need it, it has neven been checked.

**Comment by apg1980** [comment]
![grafik](https://user-images.githubusercontent.com/9514965/226890558-045f2f48-05e8-4a68-a4ed-a00907e253c7.png)

**Comment by miaulalala** [comment]
I can reproduce

**Comment by miaulalala** [comment]
> btw the dropbox on authmethod is not working.

can you elaborate?

**Comment by chrissi55** [comment]
I can confirm that the error log message 

Undefined array key "mail_smtpauthtype" at /var/www/nextcloud/apps/settings/templates/settings/admin/additional-mail.php#132

also occur when using sendmail as smtpmode!

Having this config:
https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/email_configuration.html

Sending Testmails is working but the error is logged. Also logged at startup of NC Server as well.

First time i saw this error was today after updating NC from 25.0.4 to 26.0.0

**Comment by miaulalala** [comment]
As a workaround until the fix is released, you can add a config key in your config.php:


```
  'mail_smtpauth' => 0, // if authenticationless settings
```

**Comment by miaulalala** [comment]
> didnt work after setting your recommendation same error as before

Sorry, wrong config param. There's an isset check missing in the template. I've pushed a fix, but let me test some more.

**Comment by miaulalala** [comment]
> I can confirm that the error log message
> 
> Undefined array key "mail_smtpauthtype" at /var/www/nextcloud/apps/settings/templates/settings/admin/additional-mail.php#132
> 
> also occur when using sendmail as smtpmode!
> 
> Having this config: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/email_configuration.html
> 
> my config from that is:
> 
> ```
> 'mail_smtpmode' => 'sendmail',
>   'mail_smtpsecure' => 'ssl',
>   'mail_from_address' => 'SENDER-ADDRESS',
>   'mail_domain' => 'MY.DOMAIN',
>   'mail_smtphost' => 'SMTP.MY.DOMAIN',
>   'mail_smtpport' => '465',
>   'mail_smtptimeout'  => 10,
>   'mail_smtpauth'     => false,
>   'mail_smtpauthtype' => 'LOGIN',
>   'mail_smtpname'     => '',
>   'mail_smtppassword' => '',
> ```
> 
> Sending Testmails is working but the error is logged. Also logged at startup of NC Server as well.
> 
> First time i saw this error was today after updating NC from 25.0.4 to 26.0.0

We moved to symfony mailer for 26. Unfortunately it looks like I've missed a check on the mail_smtpauthtype here. Since this param isn't used any longer, it also isn't passed from the config to the template, so setting it in the backend won't help after all.

Fix is linked above.

You can patch your instance via the instructions here if it's very intrusive to you to have that error message:
https://docs.nextcloud.com/server/latest/admin_manual/issues/applying_patch.html#patching-apps

**Comment by apg1980** [comment]
i am sorry, i am unable to understand the latest comment.
does a fix exists right now or not?

**Comment by miaulalala** [comment]
> i am sorry, i am unable to understand the latest comment. does a fix exists right now or not?

Yes, there is. You take this link: https://github.com/nextcloud/server/pull/37340.diff

download the file and follow the information in this link:

https://docs.nextcloud.com/server/latest/admin_manual/issues/applying_patch.html#patching-apps

on how to patch your instance. This will fix the issue. The fix itself will be released with 26.0.1.

**Comment by apg1980** [comment]
i have reloadet nginx and cleared my cache, it does not send an mail, issue like descrived

**Comment by apg1980** [comment]
> ![grafik](https://user-images.githubusercontent.com/9514965/226890558-045f2f48-05e8-4a68-a4ed-a00907e253c7.png)

this still appears after the Patch.

**Comment by apg1980** [comment]
Iam sorry issue is still not solved.
no email notifications can be sent out.

**Comment by miaulalala** [comment]
> > ![grafik](https://user-images.githubusercontent.com/9514965/226890558-045f2f48-05e8-4a68-a4ed-a00907e253c7.png)
> 
> this still appears after the Patch.

That looks like the settings are wrong somewhere. What's in your logs?

**Comment by apg1980** [comment]
nothing in the log.
i got this before the patch
 'mail_from_address' => 'noreply', 'mail_smtpmode' => 'smtp', 'mail_sendmailmode' => 'smtp', 'mail_domain' => 'domain.de', 'mail_smtphost' => 'IP', 'mail_smtpport' => '25', 'mail_smtpdebug' => true,

**Comment by miaulalala** [comment]
But your server offers it, at least that is what it's looking like. Symfony Mailer will try to use it if it's offered. See here:

https://github.com/symfony/symfony/issues/41997

**Comment by apg1980** [comment]
yeah thank you thats it, could it be this behaviour has changed?

**Comment by miaulalala** [comment]
> yeah thank you thats it, could it be this behaviour has changed?

We switched from SwiftMailer to Symfony Mailer as the prior is no longer developed. Symfony does behave differently than SwiftMailer in regards to STARTTLS. You can't specify it as an option, but the Mailer class will detect if a server offers it and use it.

**Comment by patvdleer** [comment]
`mail_smtpauthtype` seems to be the issue, I just did a clean install of v26 and I still have a backup of v24

**Comment by ThierryLam** [comment]
Hello,
I've done a fresh install today from the package and patch it as suggested but I still have the same error message.
Regards,
Thierry.

**Comment by hugalafutro** [comment]
Hi, 
sorry for barging in, I found this after googling the error and I am getting this error regularly in the log, however test email works ok and sends ok. Is there anything on my end I need to change in order to make the error go away or is it a matter of waiting for nc / mail app update ? I am using my own mailcow server as outgoing server using app password created in mailcow to login.
![image](https://user-images.githubusercontent.com/30209689/227892983-b53909a7-c1e5-44c5-89b5-28fa67656ce5.png)
Thanks for any info!

**Comment by MR-Welk** [comment]
> nothing in the log. i got this before the patch `
> 
> | Undefined array key "mail_smtpauthtype" at /var/www/nextcloud/apps/settings/templates/settings/admin/additional-mail.php#132 -- | --
> 
> `
> after raising debug level i got this: `Symfony\Component\Mailer\Exception\TransportException: Unable to connect with STARTTLS: stream_socket_enable_crypto(): Peer certificate CN=`maymaildomain' did not match expected CN=`10.10.20.10'`
> 
> it seems it wants startls but i have not configured starttls

**Comment by szaimen** [comment]
cc @miaulalala maybe we should put the above into some docs somewhere?

**Comment by msawyer91** [comment]
I am having the same experience as @hugalafutro in that my mail sending seems to be working just fine, but I am seeing Undefined array key "mail_smtpauthtype" at /var/www/html/apps/settings/templates/settings/admin/additional-mail.php#132 repeated in the logs.

**Comment by 404-fpv** [comment]
+1 on seeing this issue.
I have applied the patch and no longer see the error in the log, but still get an error on the mail settings > test mail.

'A problem occurred while sending the email. Please revise your settings. (Error: Email could not be sent. Check your mail server log)'

**Comment by 404-fpv** [comment]
> +1 on seeing this issue. I have applied the patch and no longer see the error in the log, but still get an error on the mail settings > test mail.
> 
> 'A problem occurred while sending the email. Please revise your settings. (Error: Email could not be sent. Check your mail server log)'

Scrap that, after applying the patch I now have a successful mail setup.
For some reason I am missing the STARTTLS option from the drop-down, however adding 'tls' in the config.php seems to have sorted that.

**Comment by andrtau** [comment]
Installed Patch, still cant send mails.
After looking in the Mailservers LOG, it seems to start TLS ignoring   'mail_smtpsecure'   => '',

IMHO TLS cant work with an IP; thus changing the IP to the Mailservers FQDN; let the Port be at 25 as it seems to autonegotiat, works with noAuth if allowed by the server (for NC ip)

Question do i need to remove the Patch before an future update or will it be overwritten anyways?


Tl;dr -> Try FQDN inplace of IP...

**Comment by miaulalala** [comment]
> > +1 on seeing this issue. I have applied the patch and no longer see the error in the log, but still get an error on the mail settings > test mail.
> > 'A problem occurred while sending the email. Please revise your settings. (Error: Email could not be sent. Check your mail server log)'
> 
> Scrap that, after applying the patch I now have a successful mail setup. For some reason I am missing the STARTTLS option from the drop-down, however adding 'tls' in the config.php seems to have sorted that.

The new underlying library doesn't support explicit STARTTLS, thus the setting was removed. If the server offers it, the connection will be automatically elevated to use STARTTLS - this is where the peer verification issue comes in.

The peer verification settings will be added to the documentation, thanks for testing.

**Comment by ilKurieita** [comment]
As mentioned above. I was able to successfully send a test email after I changed *encryption* from *ssl/tls* to *none*.

**Comment by iskraecommerce** [comment]
+1 - having the same problem, also after the patch. Using an external SMTP server based on Plesk.

**Comment by miaulalala** [comment]
> +1 - having the same problem, also after the patch. Using an external SMTP server based on Plesk.

Did you add the settings from https://github.com/nextcloud/server/issues/37329#issuecomment-1488147847 ?

**Comment by gache56** [comment]
I have the same problem with the patch in 26.0.1. But these parameters in config.php save my life and mail is now ok. The main difference is 'mail_smtpauth' => 0 instead of 'mail_smtpauth' => ''.
  'mail_smtpmode' => 'smtp',
  'mail_smtpauthtype' => '',
  'mail_smtphost' => 'XXX.XXX.XXX.XXX',
  'mail_smtpport' => '25',
  'mail_smtptimeout' => 30,
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'no-reply',
  'mail_domain' => 'mondomaine.fr',
  'mail_send_plaintext_only' => false,
  'mail_smtpauth' => 0,
  'mail_smtpname' => '',
  'mail_smtppassword' => '',
  'mail_smtpstreamoptions' => array(
    'ssl' => array(
        'allow_self_signed' => true,
        'verify_peer' => false,
        'verify_peer_name' => false
    )
  ),

**Comment by darksidewalker** [comment]
On Nextcloud 26.0.1 the STARTLS option is missing. 
Port 587 needs STARTTLS (tls). Also the loginmethod is missing.
Manually editing the config.php and adding the following lines fixes the issue:
```
'mail_smtpsecure' => 'tls'
'mail_smtpauthtype' => 'LOGIN'
```

Inside the WebUI the fields are wrong, even if the correct parameters are set on config.php.
![image](https://user-images.githubusercontent.com/84453554/235346461-2cc3a4af-ad7f-46ab-b187-5e9ff29e51f4.png)

**Comment by fdziembo** [comment]
> As mentioned above. I was able to successfully send a test email after I changed _encryption_ from _ssl/tls_ to _none_.

That worked for me!

**Comment by danielepercivaldi** [comment]
@SeanDS , I can confirm that version 26.0.1 does not fix the issue and that the configuration by @MR-Welk solved it.

Thanks.

**Comment by Lotusshaney** [comment]
I can also confirm that version 26.0.1 does not fix the issue and that the configuration by @MR-Welk solved it.

**Comment by urb4nc4rl** [comment]
Some users have worked around this issue by switching to port 465 and (presumably) SSL, which is fine.  But still a workaround.

If, like me, you REALLY need to use STARTTLS (my hosting company is blocking port 465) then you might need to do what @darksidewalker said and change your config file to:

`'mail_smtpsecure' => 'tls',`

You won't be able to do this in the web UI since it's been removed.  @miaulalala says that it's no longer necessary to have the STARTTLS option in the web UI, however Symfony auto detection wasn't working for me and was giving the error `ssl3_get_record:wrong version number`.  This implies something funky, like Symfony was trying to connect via SSL to the STARTTLS port, which it shouldn't be doing, since my understanding is that the STARTTLS protocol begins _unencrypted_ 

Anyway, the option above worked for me and seemed to force Symfony to do the 'right thing'.

**Comment by sulfatchlorid** [comment]
I had to apply the fix from https://github.com/nextcloud/server/issues/37329#issuecomment-1517630139 to be able to send emails. 
No encryption or authentication on a private IP that can be reached internally only.
Please fix it in the next release. Btw. i am running the nextcloud docker container but map the html directory to the docker host.

**Comment by punkyard** [comment]
using port 25 ans no login nor encryption is not an option for my config ..
I'm still blocked with the mail server.

**Comment by skjnldsv** [comment]
@come-nc @miaulalala 
It seems 25 without auth is still failing

**Comment by skjnldsv** [comment]
@punkyard are you saying 25 without auth is NOT working for you?
Or that you do NOT want to use 25 without auth?

**Comment by punkyard** [comment]
hi @skjnldsv 
I'm saying I'm not going to use port 25 without authentification.
Plus, port 25 is now closed, since robot attacks, which will typically use that port to do .. what ever they have to do ^^

**Comment by InFFOrmatik** [comment]
> @come-nc @miaulalala It seems 25 without auth is still failing

I can confirm that the above mentioned workaround in the config does solve the problem for the "port 25/noauth"-case too.

**Comment by btakita** [comment]
Is there any way to use the cert from protonmail-bridge, which can be obtained by running:

```
openssl s_client -starttls imap -connect 127.0.0.1:1143 -showcerts
```

See

https://stackoverflow.com/questions/57746821/getting-a-self-signed-certificate-error-with-protonbridge-and-mbsync

**Comment by danielepercivaldi** [comment]
I can confirm that version 26.0.3 does not fix the issue yet.

**Comment by unbranched** [comment]
I upgraded NC docker from 26.0.1 to 26.0.3 and the mail stopped working.
My setup is:
 ```
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'name',
  'mail_domain' => 'prettydomain.com',
  'mail_smtphost' => 'xxx.prettydomain.com',
  'mail_smtpport' => '25',
```


I confirm too that the workaround by @MR-Welk solved the issue.

**Comment by danielepercivaldi** [comment]
I can confirm that version 26.0.4 does not fix the issue yet.

**Comment by dnviti** [comment]
> I have the same problem with the patch in 26.0.1. But these parameters in config.php save my life and mail is now ok. The main difference is 'mail_smtpauth' => 0 instead of 'mail_smtpauth' => ''. 'mail_smtpmode' => 'smtp', 'mail_smtpauthtype' => '', 'mail_smtphost' => 'XXX.XXX.XXX.XXX', 'mail_smtpport' => '25', 'mail_smtptimeout' => 30, 'mail_sendmailmode' => 'smtp', 'mail_from_address' => 'no-reply', 'mail_domain' => 'mondomaine.fr', 'mail_send_plaintext_only' => false, 'mail_smtpauth' => 0, 'mail_smtpname' => '', 'mail_smtppassword' => '', 'mail_smtpstreamoptions' => array( 'ssl' => array( 'allow_self_signed' => true, 'verify_peer' => false, 'verify_peer_name' => false ) ),

'mail_smtpauthtype' => 0
It worked...
How in hell they passed a pull request with such a disruptive USELESS change?
It destroyed my shares workflow for 2 weeks and i had to contact all my users (500+ instance) and explain to averyone how to manually share with link (which also is missing user share traceability lowering security).
My superior also now wants to change to onedrive cause of this problem.
Please try to pay more attention next time or you'll lose users...

**Comment by danielepercivaldi** [comment]
Hello @come-nc .
If you set up email parameters as shown in figure, sending email fails. It starts working with configuration suggested by @MR-Welk .

Thanks.

![image](https://github.com/nextcloud/server/assets/8720648/e9a04db2-9f60-4e79-b5ba-f1fcafbd9167)

**Comment by sheggy-rin** [comment]
I have same problem. 
On Nextcloud 26.0.4, i'm getting error with the following settings
![new](https://github.com/nextcloud/server/assets/119958013/338de5af-046a-4349-b89f-abedef363a53)
but, if i set up another  server withoutt checkbox "authentication required" on same port, its worked
on Nextcloud 24, its ok
![old](https://github.com/nextcloud/server/assets/119958013/c609f75a-82cd-47d4-bbbb-cfc7f0ec7a64)

'mail_smtpdebug' => true  also does not work

**Comment by come-nc** [comment]
> Hello @come-nc . If you set up email parameters as shown in figure, sending email fails. It starts working with configuration suggested by @MR-Welk .

You use localhost to connect to your mailserver and most likely your SSL certificate is not valid for localhost. Your mail server advertise support for STARTTLS and with the new library we use it is not possible to avoid it in this case. So either you need a certificate valid for localhost, or you need to use the domain matching the certificate instead of localhost, or you need to add the configuration suggested by MR-Welk which ignores SSL validation.
We could see about adding either a UI to support this configuration, or maybe force it for localhost and make it a special case, but that is a subject for another ticket, please open it if you want to follow on this subject.

> I have same problem.
> On Nextcloud 26.0.4, i'm getting error with the following settings

@sheggy-rin This is not the same problem, you are not using localhost and you say that using STARTTLS does work with your mailserver. I suspect it might be an encryption configuration which is not supported anymore by the new library but it might also be a bug, you can open a bug ticket on the subject if you need, but in this case provide as many details as possible, especially look if you get log entries related to the connection attempt in the logs. Maybe enable debug logging before attempting the connection to get more logs and disable it after.

**Comment by danielepercivaldi** [comment]
Hi @come-nc 
I can confirm that assigning an SSL certificate to the local mail server and using an FQDN in replacement of "localhost" sending email works.

Thanks.

**Comment by Baltimorepc** [comment]
i have this issue in the AIO V28 it was working fine before this 28 update

**Comment by ayoahha** [comment]
Yes, so am i, same f**** issue with 29.0.0.2
Trying to solve since 2 days now

No reco here are functioning

In the logs (debug) i see :
```
TransportException
Connection could not be established with host "smtp.domain.com:587": stream_socket_client(): Unable to connect to smtp.domain.com:587 (Connection timed out)
Sending mail to "Array ( [address@gmail.com] => ncc_login ) " with subject ".........................." failed
```
I do not want to open 25 port
Tried also with 465, but no success

Can we reopen this please ?
Where can we find other logs ?

Here is my config.php file for email :

```
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'info',
  'mail_domain' => 'domain.com',
  'mail_smtpport' => '587',
  'mail_smtphost' => 'smtp.domain.com',
  'mail_smtpstreamoptions' => 
  array (
    'ssl' => 
    array (
      'allow_self_signed' => true,
      'verify_peer' => false,
      'verify_peer_name' => false,
    ),
  ),
  'mail_smtptimeout' => 30,
  'mail_smtpauth' => 1,
  'mail_smtpname' => 'info@domain.com',
  'mail_smtppassword' => 'PASSWD',
  'mail_smtpdebug' => true,
  'loglevel' => 0,
```

Many thanks for those who can help

**Comment by miaulalala** [comment]
> Yes, so am i, same f**** issue with 29.0.0.2 Trying to solve since 2 days now
> 
> No reco here are functioning
> 
> In the logs (debug) i see :
> 
> ```
> TransportException
> Connection could not be established with host "smtp.domain.com:587": stream_socket_client(): Unable to connect to smtp.domain.com:587 (Connection timed out)
> Sending mail to "Array ( [address@gmail.com] => ncc_login ) " with subject ".........................." failed
> ```
> 
> I do not want to open 25 port Tried also with 465, but no success
> 
> Can we reopen this please ? Where can we find other logs ?
> 
> Here is my config.php file for email :
> 
> ```
>   'mail_smtpmode' => 'smtp',
>   'mail_sendmailmode' => 'smtp',
>   'mail_from_address' => 'info',
>   'mail_domain' => 'domain.com',
>   'mail_smtpport' => '587',
>   'mail_smtphost' => 'smtp.domain.com',
>   'mail_smtpstreamoptions' => 
>   array (
>     'ssl' => 
>     array (
>       'allow_self_signed' => true,
>       'verify_peer' => false,
>       'verify_peer_name' => false,
>     ),
>   ),
>   'mail_smtptimeout' => 30,
>   'mail_smtpauth' => 1,
>   'mail_smtpname' => 'info@domain.com',
>   'mail_smtppassword' => 'PASSWD',
>   'mail_smtpdebug' => true,
>   'loglevel' => 0,
> ```
> 
> Many thanks for those who can help

You've got a connection timeout, not a code issue. Can you PING the domain? Is port 578 open? Can you connect from any other client and send emails?

**Comment by Baltimorepc** [comment]
I been dealing with this for a while it be nice for them to get it to work . Right now I can’t even update AIO as all the apps don’t work and crashes so I am still on v 28


Sent from Yahoo Mail for iPhone


On Tuesday, June 25, 2024, 6:20 AM, Anna ***@***.***> wrote:





Yes, so am i, same f**** issue with 29.0.0.2 Trying to solve since 2 days now

No reco here are functioning

In the logs (debug) i see :
TransportException
Connection could not be established with host "smtp.domain.com:587": stream_socket_client(): Unable to connect to smtp.domain.com:587 (Connection timed out)
Sending mail to "Array ( ***@***.*** => ncc_login ) " with subject ".........................." failed

I do not want to open 25 port Tried also with 465, but no success

Can we reopen this please ? Where can we find other logs ?

Here is my config.php file for email :
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'info',
  'mail_domain' => 'domain.com',
  'mail_smtpport' => '587',
  'mail_smtphost' => 'smtp.domain.com',
  'mail_smtpstreamoptions' => 
  array (
    'ssl' => 
    array (
      'allow_self_signed' => true,
      'verify_peer' => false,
      'verify_peer_name' => false,
    ),
  ),
  'mail_smtptimeout' => 30,
  'mail_smtpauth' => 1,
  'mail_smtpname' => ***@***.***',
  'mail_smtppassword' => 'PASSWD',
  'mail_smtpdebug' => true,
  'loglevel' => 0,

Many thanks for those who can help


You've got a connection timeout, not a code issue. Can you PING the domain? Is port 578 open? Can you connect from any other client and send emails?

**Comment by SuperdukeGates** [comment]
On 2024/6/23 02:17, ayoahha wrote:
    
    
      
      Yes, so am i, same f**** issue with 29.0.0.2
        Trying to solve since 2 days now
      No reco here are functioning
      In the logs (debug) i see :
      TransportException
Connection could not be established with host "smtp.domain.com:587": stream_socket_client(): Unable to connect to smtp.domain.com:587 (Connection timed out)
Sending mail to "Array ( ***@***.*** => ncc_login ) " with subject ".........................." failed

      I do not want to open 25 port
        Tried also with 465, but no success
      Can we reopen this please ?
        Where can we find other logs ?
      Here is my config.php file for email :
        'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_from_address' => 'info',
  'mail_domain' => 'domain.com',
  'mail_smtpport' => '587',
  'mail_smtphost' => 'smtp.domain.com',
  'mail_smtpstreamoptions' => 
  array (
    'ssl' => 
    array (
      'allow_self_signed' => true,
      'verify_peer' => false,
      'verify_peer_name' => false,
    ),
  ),
  'mail_smtptimeout' => 30,
  'mail_smtpauth' => 1,
  'mail_smtpname' => ***@***.***',
  'mail_smtppassword' => 'PASSWD',
  'mail_smtpdebug' => true,
  'loglevel' => 0,

      Many thanks for those who can help
    
    If your mail server supports submissions (ie. smtps) , then you
    could try port 465.

**Comment by Baltimorepc** [comment]
I finally got mine working 
Did you go into the config.php and add the needed rows to allow self signed certs 


Sent from Yahoo Mail for iPhone


On Wednesday, June 26, 2024, 7:25 AM, SuperdukeGates ***@***.***> wrote:




 
 
 
 
 On 2024/6/23 02:17, ayoahha wrote:
 
 
 
 Yes, so am i, same f**** issue with 29.0.0.2
 Trying to solve since 2 days now
 No reco here are functioning
 In the logs (debug) i see :
 TransportException
Connection could not be established with host "smtp.domain.com:587": stream_socket_client(): Unable to connect to smtp.domain.com:587 (Connection timed out)
Sending mail to "Array ( ***@***.*** => ncc_login ) " with subject ".........................." failed

 I do not want to open 25 port
 Tried also with 465, but no success
 Can we reopen this please ?
 Where can we find other logs ?
 Here is my config.php file for email :
 'mail_smtpmode' => 'smtp',
 'mail_sendmailmode' => 'smtp',
 'mail_from_address' => 'info',
 'mail_domain' => 'domain.com',
 'mail_smtpport' => '587',
 'mail_smtphost' => 'smtp.domain.com',
 'mail_smtpstreamoptions' => 
 array (
 'ssl' => 
 array (
 'allow_self_signed' => true,
 'verify_peer' => false,
 'verify_peer_name' => false,
 ),
 ),
 'mail_smtptimeout' => 30,
 'mail_smtpauth' => 1,
 'mail_smtpname' => ***@***.***',
 'mail_smtppassword' => 'PASSWD',
 'mail_smtpdebug' => true,
 'loglevel' => 0,

 Many thanks for those who can help
 
 If your mail server supports submissions (ie. smtps) , then you
 could try port 465.
